### PR TITLE
[NaiveSolver] Check allTrades against clearing prices

### PR DESCRIPTION
**Pull request by fleupold**
The NaiveSolver has been making some bad trades ([context](https://cowservices.slack.com/archives/C036JAGRQ04/p1676455072588869)).

The root cause is that we didn't check if the price we execute liquidity orders at (their limit price) is respecting the uniform clearing price and filter solutions which don't. To address this, this PR does four things:
1. Fix the unit test we had in place for this (didn't survive the introduction of OrderClass::Liquidity) making it catch the issue
2. Consider `all_trades` instead of just `user_trades` when checking valid solutions in the Naive Solver
3. Additional check to see if the order's limit price is in line with the uniform clearing price (as liquidity orders' execution prices are overridden to be their limit price, thus always passing the existing check in that loop).
4. Add another test to check limit orders that have a different price after fees are not filtered out by this (applying the fee only makes their price worse and does not "overshoot" the UCP unlike the liquidity order in the example)

### Test Plan

Existing "corrected" unit test (fails without the other changes).

### Release notes

Allows re-enabling the NaiveSolver

### CODE REVIEW

**Review by nlordell — APPROVED** [review]
Changes look correct to me.

File: crates/solver/src/solver/naive_solver.rs [hunk 1]
```diff
@@ -458,4 +465,76 @@ mod tests {
...
+                address: addr!("210a97ba874a8e279c95b350ae8ba143a143c159"),
+                tokens,
+                reserves: (1_000_000_000_000, 1_000_000_000_000),
+                fee: Ratio::new(3, 1000),
+                settlement_handling: amm_handler,
+            },
+        };
+
+        let settlement = &settle(SlippageContext::default(), orders, liquidity)[0];
+        assert_eq!(
```
Comment (vkgnosis) [inline crates/solver/src/solver/naive_solver.rs:522]: Could also check that both orders really got settled.

**Review by vkgnosis — APPROVED** [review]

**Comment by vkgnosis** [comment]
CI is failing because of formatting. We're using nightly rustfmt which specifically changes how imports are formatted. You can install it with `rustup install nightly` and use it with `cargo +nightly format`. Or one of us can push the formatting fix and merge.

File: crates/solver/src/solver/naive_solver/multi_order_solver.rs [hunk 2]
```diff
@@ -316,16 +316,37 @@ fn is_valid_solution(solution: &Settlement) -> bool {
...
+        // E.g. liquidity orders may have a different executed price.
+        match (
+            solution
+                .encoder
+                .clearing_prices()
+                .get(&order.buy_token)
+                .map(|clearing_sell_price| clearing_sell_price.checked_mul(sell_token_price)),
+            solution
+                .encoder
+                .clearing_prices()
```
Comment (MartinquaXD) [inline crates/solver/src/solver/naive_solver/multi_order_solver.rs:342]: nit: a little less verbose
```suggestion
        let clearing_prices = solution.encoder.clearing_prices();
        match (
            clearing_prices
                .get(&order.buy_token)
                .map(|clearing_sell_price| clearing_sell_price.checked_mul(sell_token_price)),
            clearing_prices
```

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**